federation/pkg/kubefed/init/init.go

```diff
@@ -236,6 +248,14 @@ func (i *initFederation) Run(cmdOut io.Writer, config util.AdminConfig) error {
 	cmName := fmt.Sprintf("%s-controller-manager", i.commonOptions.Name)
 	cmKubeconfigName := fmt.Sprintf("%s-kubeconfig", cmName)
 
+	var dnsProviderConfigBytes []byte
```

Comment:
(no action required) Why not read the file in ``Complete`` rather than stat'ing there and reading here?